### PR TITLE
Added function to plot 2d array data from an AnalogSignal

### PR DESCRIPTION
**Pull request by mdenker**
This is a first function to plot 2D array data on a grid. It's code originally used for Denker et al., 2016, but already is reasonably generic. I think there are a number of things one could improve and make it even more generic, but it's a start.

### CODE REVIEW

File: viziphant/array_data.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+"""
+This module contains functions to plot activity data from two-dimensional
+recording arrays.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import neo
+
```
Comment (Qkley) [inline viziphant/array_data.py]: Quantities package is used in doctest
```suggestion
import quantities as pq
```

File: viziphant/array_data.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    Example
+    =======
+    # Create fake data for a 4x25 electrode array, that is the array is 4 
+    # electrodes high and 25 wide. Electrode 1 is bottom left, electrode 100 
+    # is top right. We set channel to a high value. Also, for simplicity, we'll
+    # create only one sample of the AnalogSignal.
+    >>> grid_data = np.random.random((1,100))
+    >>> grid_data[0,3] = 3
+    >>> grid_data_obj = neo.AnalogSignal(
+            grid_data*pq.mV, t_start=0*pq.s, sampling_rate=1000*pq.Hz)
```
Comment (Qkley) [inline viziphant/array_data.py]: Multiline formatting for doctests following convention
```suggestion
    ...     grid_data*pq.mV, t_start=0*pq.s, sampling_rate=1000*pq.Hz)
```

File: viziphant/array_data.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    
+    # Create corresponding directions in the direction 0 rad
+    direction_data = np.exp(np.complex(0,1)*np.random.random((1,100))*.5)
```
Comment (Qkley) [inline viziphant/array_data.py]: A fix to a part of the doctest
```suggestion
    >>> direction_data = np.exp(np.complex(0,1)*np.random.random((1,100))*.5)
```

File: viziphant/array_data.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    >>> direction_data_obj = neo.AnalogSignal(
+            direction_data*pq.dimensionless,
```
Comment (Qkley) [inline viziphant/array_data.py]:
```suggestion
    ...     direction_data*pq.dimensionless, 
```

File: viziphant/array_data.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    >>> grid_data = np.random.random((1,100))
+    >>> grid_data[0,3] = 3
+    >>> grid_data_obj = neo.AnalogSignal(
+            grid_data*pq.mV, t_start=0*pq.s, sampling_rate=1000*pq.Hz)
+    
+    # Create corresponding directions in the direction 0 rad
+    direction_data = np.exp(np.complex(0,1)*np.random.random((1,100))*.5)
+    >>> direction_data_obj = neo.AnalogSignal(
+            direction_data*pq.dimensionless, 
+            t_start=0*pq.s, sampling_rate=1000*pq.Hz)
```
Comment (Qkley) [inline viziphant/array_data.py]:
```suggestion
    ...     t_start=0*pq.s, sampling_rate=1000*pq.Hz)
```

File: viziphant/array_data.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    >>> cmap = plt.cm.get_cmap('hsv')
+    
+    # Plot the data
+    >>> plot_array(
+            plt.gca(), (4,25), grid_data_obj, direction_data_obj,
```
Comment (Qkley) [inline viziphant/array_data.py]:
```suggestion
    ...     plt.gca(), (4,25), grid_data_obj, direction_data_obj, 
```

File: viziphant/array_data.py [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    direction_data = np.exp(np.complex(0,1)*np.random.random((1,100))*.5)
+    >>> direction_data_obj = neo.AnalogSignal(
+            direction_data*pq.dimensionless, 
+            t_start=0*pq.s, sampling_rate=1000*pq.Hz)
+    >>> cmap = plt.cm.get_cmap('hsv')
+    
+    # Plot the data
+    >>> plot_array(
+            plt.gca(), (4,25), grid_data_obj, direction_data_obj, 
+            arrow_scale_factor=.7, summary_arrow_scale_factor=.5,
```
Comment (Qkley) [inline viziphant/array_data.py]:
```suggestion
    ...     arrow_scale_factor=.7, summary_arrow_scale_factor=.5, 
```

File: viziphant/array_data.py [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    >>> direction_data_obj = neo.AnalogSignal(
+            direction_data*pq.dimensionless, 
+            t_start=0*pq.s, sampling_rate=1000*pq.Hz)
+    >>> cmap = plt.cm.get_cmap('hsv')
+    
+    # Plot the data
+    >>> plot_array(
+            plt.gca(), (4,25), grid_data_obj, direction_data_obj, 
+            arrow_scale_factor=.7, summary_arrow_scale_factor=.5, 
+            color_map=cmap)
```
Comment (Qkley) [inline viziphant/array_data.py]:
```suggestion
    ...     color_map=cmap)
```

File: viziphant/array_data.py [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+        # rotate clockwise
+        rotation = 2.*np.pi-rotation
+
+        # affine transform
+        X = X - (dimension[0]/2-0.5) #4.5
+        Y = Y - (dimension[1]/2-0.5) #4.5
+        Xr = X * np.cos(rotation) - Y * np.sin(rotation)
+        Yr = X * np.sin(rotation) + Y * np.cos(rotation)
+        X = Xr + (dimension[0]/2-0.5)
+        Y = Yr + (dimension[0]/2-0.5)
```
Comment (Qkley) [inline viziphant/array_data.py]: Should this be a 'dimension[1]' instead of 'dimension[0]'?
```suggestion
        Y = Yr + (dimension[1]/2-0.5)
```

Comment (mdenker) [reply viziphant/array_data.py]: Good catch, however, actually it must be 1 for X and 0 for Y here and further up....

File: viziphant/array_data.py (hunk 1 above)
Comment (mdenker) [reply viziphant/array_data.py]: So for now, I put the quantities library in the doc test itself, but this should be discussed again.

Comment (dizcza) [reply viziphant/array_data.py]: Answering your first question,
we can do `pip install viziphant[test]` or `[all]` if we define the appropriate flag in `setup.py`.
but now I noticed a larger problem - we don't have requirements at all - neither `requirements.txt` nor in `setup.py`.
-_-

**Pull request merged**